src/app/CMakeLists.txt

```diff
@@ -77,31 +77,95 @@ set_source_files_properties(
 if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     # substitute @EXECUTABLE@ in dist/mac/Info.plist
     get_target_property(EXECUTABLE qbt_app OUTPUT_NAME)
-    configure_file(${qBittorrent_SOURCE_DIR}/dist/mac/Info.plist
-        ${qBittorrent_BINARY_DIR}/dist/mac/pregen.plist @ONLY)
+    configure_file("${qBittorrent_SOURCE_DIR}/dist/mac/Info.plist"
+        "${qBittorrent_BINARY_DIR}/dist/mac/pregen.plist" @ONLY)
     file(GENERATE
-        OUTPUT ${qBittorrent_BINARY_DIR}/dist/mac/Info.plist
-        INPUT ${qBittorrent_BINARY_DIR}/dist/mac/pregen.plist
+        OUTPUT "${qBittorrent_BINARY_DIR}/dist/mac/Info.plist"
+        INPUT "${qBittorrent_BINARY_DIR}/dist/mac/pregen.plist"
     )
     set_target_properties(qbt_app PROPERTIES
         MACOSX_BUNDLE ON
         MACOSX_BUNDLE_BUNDLE_NAME "${EXECUTABLE}"
-        MACOSX_BUNDLE_INFO_PLIST ${qBittorrent_BINARY_DIR}/dist/mac/Info.plist
+        MACOSX_BUNDLE_INFO_PLIST "${qBittorrent_BINARY_DIR}/dist/mac/Info.plist"
     )
     target_sources(qbt_app PRIVATE
-        ${QT_TRANSLATIONS}
-        ${qBittorrent_SOURCE_DIR}/dist/mac/qt.conf
-        ${qBittorrent_SOURCE_DIR}/dist/mac/qBitTorrentDocument.icns
-        ${qBittorrent_SOURCE_DIR}/dist/mac/qbittorrent_mac.icns
+        ${QT_TRANSLATIONS_LIST}
+        "${qBittorrent_SOURCE_DIR}/dist/mac/qt.conf"
+        "${qBittorrent_SOURCE_DIR}/dist/mac/qBitTorrentDocument.icns"
+        "${qBittorrent_SOURCE_DIR}/dist/mac/qbittorrent_mac.icns"
     )
 elseif (CMAKE_SYSTEM_NAME STREQUAL "Windows")
     set_target_properties(qbt_app PROPERTIES WIN32_EXECUTABLE ON)
     if (MINGW)
-        target_sources(qbt_app PRIVATE ${qBittorrent_SOURCE_DIR}/src/qbittorrent_mingw.rc)
+        target_sources(qbt_app PRIVATE "${qBittorrent_SOURCE_DIR}/src/qbittorrent_mingw.rc")
     else()
-        target_sources(qbt_app PRIVATE ${qBittorrent_SOURCE_DIR}/src/qbittorrent.rc)
+        target_sources(qbt_app PRIVATE "${qBittorrent_SOURCE_DIR}/src/qbittorrent.rc")
+    endif()
+    target_sources(qbt_app PRIVATE "${qBittorrent_SOURCE_DIR}/src/qbittorrent.exe.manifest")
+
+    # Copy DLLs to the build folder after the build, so that the executable can be run from there.
+    # When using vcpkg, everything is automatically copied except for some stuff handled by windeployqt;
+    # this is no problem because the copy_if_different command is idempotent in this case
+    # TODO: works for MSVC, what about MinGW?
+    # TODO: This can probably be greatly simplified by using TARGET_RUNTIME_DLLS instead, in CMake >= 3.21
+    if (MSVC_RUNTIME_DYNAMIC)
```

Comment:
Seems you should check TYPE property for each dependency lib instead.

Reply:
@glassez 

Hmm, do we really want to support/endorse the use case of arbitrary mixes of static/dynamic versions of the dependent libraries? Does that even work properly in all cases? I understand supporting dynamic runtime + static libraries (in addition to both dynamic), but this seems a bit excessive.

With that in mind, I propose testing just Qt (for example), and assuming that its type is the same for all other dependencies. What do you think?

Reply:
@glassez

Another relevant point to consider: https://docs.microsoft.com/en-us/cpp/error-messages/tool-errors/linker-tools-warning-lnk4098?view=msvc-160.

> You should compile all source files to use the same run-time library.

Although it is technically possible and there are documented workarounds, thanks to this recommendation in the docs it seems to me we can simply tell the user they can either do fully static or fully dynamic, but not a mix of both. Then we could add `(use this when building with dynamic libraries)` to the description of the `MSVC_RUNTIME_DYNAMIC` option (if we do that, then additionally we should probably rename that option to `MSVC_DYNAMIC_BUILD` or something).

Thoughts?

Reply:
Generally, there is nothing bad, incorrect, or unacceptable about mixing static and dynamic libraries (i.e., linking an executable file to multiple libraries of different linking types). Even indirect linking to different versions of the same library is quite acceptable (for example, this happens when you link qBittorrent with dynamic Qt and zlib, while the Qt itself is statically linked with zlib). We are only talking about some specific cases, for example, MSVC runtime library. So I'm totally against intentionally limiting it at project level.
>With that in mind, I propose testing just Qt (for example), and assuming that its type is the same for all other dependencies. What do you think?

On the contrary, it may be a good idea to link with it dynamically (due to its large size), but statically with other (smaller) libraries. In any case, quite a lot of valid reasons for such mixing can be imagined.

Reply:
@glassez 

> So I'm totally against intentionally limiting it at project level.

OK, you've convinced me that we should not limit it (or even suggest that it should be better limited) - after all, there is no reason to make it unnecessarily difficult for others to add support for potentially useful workflows for them. But how about just considering it out-of-scope for now? If someone wants better support for that, they can implement it themselves later on (or I may even end up doing it myself at a later time).